File: clients/client-java/src/test/java/com/datastrato/gravitino/client/TestTokenAuthProvider.java
/*
 * Copyright 2023 Datastrato Pvt Ltd.
 * This software is licensed under the Apache License version 2.
 */
package com.datastrato.gravitino.client;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;
import static org.junit.jupiter.api.Assertions.assertTrue;

import com.datastrato.gravitino.auth.AuthConstants;
import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.util.Base64;
import org.junit.jupiter.api.Test;

public class TestTokenAuthProvider {
  @Test
  public void testAuthentication() throws IOException {
    try (AuthDataProvider provider = new TokenAuthProvider("test:token")) {
      assertTrue(provider.hasTokenData());
      String token = new String(provider.getTokenData(), StandardCharsets.UTF_8);
      assertTrue(token.startsWith(AuthConstants.AUTHORIZATION_TOKEN_HEADER));
      String tokenString =
          new String(
              Base64.getDecoder()
                  .decode(
                      token
                          .substring(AuthConstants.AUTHORIZATION_TOKEN_HEADER.length())
                          .getBytes(StandardCharsets.UTF_8)),
              StandardCharsets.UTF_8);
      assertEquals("test:token", tokenString);
    }
  }

  @Test
  public void testNull() {
    assertThrows(NullPointerException.class, () -> new TokenAuthProvider(null));
  }
}

File: clients/client-python/tests/unittests/test_token_auth_provider.py
"""
Copyright 2024 Datastrato Pvt Ltd.
This software is licensed under the Apache License version 2.
"""

import base64
import os
import unittest

from gravitino.auth.auth_constants import AuthConstants
from gravitino.auth.auth_data_provider import AuthDataProvider
from gravitino.auth.simple_auth_provider import SimpleAuthProvider
from gravitino.auth.token_auth_provider import TokenAuthProvider
from gravitino.exceptions.gravitino_runtime_exception import GravitinoRuntimeException


class TestSimpleAuthProvider(unittest.TestCase):

    def test_auth_provider(self):
        mock_token = "this is test"
        token_provider: TokenAuthProvider = TokenAuthProvider(mock_token)
        self.assertTrue(token_provider.has_token_data())
        decode_token = token_provider.get_token_data().decode("utf-8")[
            len(f"{AuthConstants.AUTHORIZATION_TOKEN_HEADER}") :
        ]
        self.assertEqual(mock_token, base64.b64decode(decode_token).decode("utf-8"))

    def test_null(self):
        with self.assertRaises(GravitinoRuntimeException):
            token = ""
            token_provider: TokenAuthProvider = TokenAuthProvider(token)

        with self.assertRaises(TypeError):
            token_provider: TokenAuthProvider = TokenAuthProvider()
